library/src/main/java/com/google/android/material/motion/runtime/TargetScope.java

```diff
@@ -118,24 +132,42 @@ private int getDetailedState() {
     return state;
   }
 
+  private void removeNamedPlan(NamedPlan plan, String name, Performer.NamedPlanPerformance performer) {
+    if (performer != null) {
+      performer.removePlan(plan, name);
+      namedCache.remove(name);
+    }
+  }
+
+  private Performer.NamedPlanPerformance getNamedPerformer(NamedPlan plan, String name, Object target) {
```

Comment:
I tried removing this method & just always doing the lookup in the `namedCache`, but the unit tests failed with errors like:
```
com.google.android.material.motion.runtime.SchedulerTest > testRemoveNamedPlanThatWasNeverAdded[Nexus_6_API_23(AVD) - 6.0] FAILED 
	java.lang.NullPointerException: Attempt to invoke interface method 'void com.google.android.material.motion.runtime.Performer$NamedPlanPerformance.addPlan(com.google.android.material.motion.runtime.NamedPlan, java.lang.String)' on a null object reference
	at com.google.android.material.motion.runtime.TargetScope.commitAddNamedPlan(TargetScope.java:96)
:library:connectedDebugAndroidTest FAILED
```
I was thinking `getPerformerForName()` would be useful to wrap this logic:
```
1. Try to look up the performer in namedCache
2. If I cant find it in namedCache, try to create it
3. If I have to create the performer, remember to put it into namedCache
```
Are you suggesting that the `commitAddNamedPlan` method use logic as described above? And, `commitRemoveNamedPlan` would only ever try to look up the performer in `namedCache` ?